src/EVMBlackjack.sol

```diff
@@ -0,0 +1,337 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+import {Chip} from "../src/Chip.sol";
+
+contract EVMBlackjack {
+    /*//////////////////////////////////////////////////////////////
+    //  Events
+    //////////////////////////////////////////////////////////////*/
+
+    event BetPlaced(address indexed player, uint16 betSize);
+    event PlayerCardDealt(address indexed player, uint8 card, uint8 handIndex);
+    event DealerCardDealt(address indexed player, uint8 card);
+    event InsuranceTaken(address indexed player, bool take);
+    event PlayerActionTaken(address indexed player, Action action);
+    event PlayerBust(address indexed player);
+    event DealerBust(address indexed player);
+    event PayoutsHandled(address indexed player, uint16 playerPayout, uint16 dealerPayout);
+
+    /*//////////////////////////////////////////////////////////////
+    //  Errors
+    //////////////////////////////////////////////////////////////*/
+
+    error InvalidStateTransition();
+    error InvalidBetSize(uint16 betSize);
+
+    /*//////////////////////////////////////////////////////////////
+    //  Data Structures
+    //////////////////////////////////////////////////////////////*/
+
+    enum State {
+        READY_FOR_BET,
+        WAITING_FOR_RANDOMNESS,
+        READY_FOR_INSURANCE,
+        READY_FOR_PLAYER_ACTION,
+        DEALER_ACTION,
+        PAYOUTS
+    }
+
+    enum Action {
+        NO_ACTION,
+        SPLIT_ACES,
```

Comment:
can we drop split_aces, and just use split? the logic for what happens when you split aces can be conditioned on the contents of the hand. 

also, what happens if a player that has pair aces sends SPLIT instead of SPLIT_ACES?